src/exercises/exerciseTypes/orderWords/OrderWords.js

```diff
@@ -1,774 +1,1138 @@
-        api.getSmallerContext(originalContext, bookmarkWord,
-          exerciseLang, MAX_CONTEXT_LENGTH, (apiCandidateSubSent) => {
-            let shorterContext = JSON.parse(apiCandidateSubSent);
-            prepareExercise(shorterContext, isCurrentSentenceTooLong, isHandleLongSentences, exerciseStartTime);
-          });
-      }
-      else {
-        console.log("Using default context.");
-        prepareExercise(originalContext, isCurrentSentenceTooLong, isHandleLongSentences, exerciseStartTime);
-      }
-  }
-
-  function prepareExercise(exerciseContext, isSentenceTooLong, isHandlingLongSentences, startTime) {
-    console.log("CONTEXT: '" + exerciseContext + "'");
-    exerciseContext = exerciseContext.trim();
-    console.log("CONTEXT AFTER TRIM: '" + exerciseContext + "'");
-    console.log("Getting Translation for ->" + exerciseContext);
-    
-    setExerciseContext(exerciseContext);
-    
-    let originalContext = bookmarksToStudy[0].context.trim();
-
-    api
-      .basicTranlsate(
-        exerciseLang,
-        localStorage.native_language,
-        exerciseContext
-      )
-      .then((response) => response.json())
-      .then((data) => {
-        let translatedContext = data["translation"];
-        // Line below is used for development with no API key (translatedContext is Null)
-        if (!translatedContext) { translatedContext = exerciseContext; }
-        if (exerciseContext.length < originalContext.length){
-          let startPos = originalContext.search(exerciseContext);
-          let contextLen = originalContext.length;
-          setTextBeforeTranslatedText(originalContext.slice(0,startPos));
-          setTextAfterTranslatedText(originalContext.slice(startPos+exerciseContext.length,contextLen));
-        }        
-        setTranslatedText(translatedContext);
-        createConfusionWords(exerciseContext, translatedContext, 
-          isSentenceTooLong, isHandlingLongSentences, startTime);
-
-      })
-      .catch(() => {
-        let translationError = "Error retrieving the translation.";
-        setTranslatedText(translationError);
-        console.log("could not retreive translation");
-        setConfuseWords([]);
-        setWordsReferenceStatus([""]);
+
+
+    if (IS_DEBUG) console.log("Running ORDER WORDS EXERCISE");
+
+    function _removeEmptyTokens(tokenList) {
+        // In some instance, there will be punctuation in the middle, which
+        // results in trailing spaces. The loop below ensures those get removed.
+        return tokenList.filter((token) => token !== "");
+    }
+
+    function _getWordsInSentence(sentence) {
+        let wordsForExercise = tokenize(removePunctuation(sentence));
+        // A lot of  articles start with a dash. ( - )
+        if (wordsForExercise[0] === "-") wordsForExercise = wordsForExercise.splice(1)
+        return _removeEmptyTokens(wordsForExercise);
+    }
+
+    function _getWordsFromWordProps(wordPropList) {
```

Comment:
can we add a comment on what's this doing? I don't understand what are the "wordProps"?

Reply:
That method takes a list of wordProps and returns a list of strings (the words themselves). The wordProps is the data structure that I use for the word items (a dictionary) initialized in what I now named _initializedWordProps (for consistency). That method also describes each property tracked.